devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusBuild.java

```diff
@@ -78,6 +78,30 @@ public FileCollection getClasspath() {
                 .plus(mainSourceSet.getResources());
     }
 
+    @Input
+    public Map<String, String> getQuarkusBuildSystemProperties() {
+        Map<String, String> quarkusSystemProperties = new HashMap<>();
+        for (Map.Entry<Object, Object> systemProperty : System.getProperties().entrySet()) {
+            if (systemProperty.getKey() != null && systemProperty.getKey() instanceof String
+                    && systemProperty.getValue() instanceof String
+                    && ((String) systemProperty.getKey()).startsWith("quarkus.")) {
+                quarkusSystemProperties.put((String) systemProperty.getKey(), (String) systemProperty.getValue());
```

Comment:
Additional question: is it mandatory that the value is a string?

Reply:
Not necessarily but it must be `Serializable`

Reply:
I wonder why we make it mandatory for the values then? I don't see a lot of value in forcing it for the values but I might miss something :)

Reply:
Maybe just test if `systemProperty.getValue() instanceof Serializable` should be enough, since we don't care about the value here

Reply:
right.